### PR TITLE
Candlestick chart overlays with the line graph

### PR DESCRIPTION
**Pull request by arealclimber**
### DEVELOP
- [x] CandlestickChart
- [x] TradingChartSwitch 
- [x] TradingView 

### CHECK LIST

- [Naming convention](https://github.com/CAFECA-IO/WorkGuidelines/blob/main/technology/coding-convention/naming-convention.md) verification: checked
- Coding style verification: checked
- new Library: 1
   - victory: 36.6.8
- new Class / Component: 0
- new loop: 4
   - src/components/candlestick_chart/candlestick_chart.tsx (43, 70, 105, 391)
- new recursive: 0
- high risk: 0
- new sql: 0

### UML

- [Sequence Diagram 001 (interaction between user, frontend and backend)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00001.md)
- [Sequence Diagram 002 (background mechanism)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00002.md)
- [Component Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCP00001.md)
- [Class Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCD00001.md)
- [Global context document](https://www.notion.so/Global-Context-public-d93ac7c941c94fb6b173d34a7bc4e959)
- [Table of interface](https://github.com/CAFECA-IO/EventReport/blob/main/meeting/20230106_tbd_mvp.md)

### CODE REVIEW

File: src/components/candlestick_chart/candlestick_chart.tsx [hunk 1]
```diff
@@ -433,59 +1105,481 @@ export default function CandlestickChart({
   // console.log('showPosition state:', showPositionOnChart);
   // console.log('display option:', displayedPosition.annotations.yaxis[0].label.text);
 
-  const [dataSample, setDataSample] = useState({
+  const [lineChart, setLineChart] = useState({
+    options: lineChartOptions,
+  });
+
+  const [candleChart, setCandleChart] = useState({
```
Comment (arealclimber) [inline src/components/candlestick_chart/candlestick_chart.tsx:1112]: - 多餘的程式碼先註解起來

File: src/components/candlestick_chart/candlestick_chart.tsx [hunk 2]
```diff
@@ -433,59 +1105,481 @@ export default function CandlestickChart({
...
+    <>
+      {/* w-2/3 xl:w-4/5 */}
+      {/* TODO: Temporary adjustment of chart size */}
+      <div className="-ml-5" style={{width: '70%'}}>
+        {/* <VictoryCandlestick
+          // candleWidth={55}
+          data={sampleDataDates}
+          // data={marketCtx.candlestickChartData ? [...marketCtx.candlestickChartData] : []}
+        /> */}
+        {!marketCtx.candlestickChartData !== null ? (
```
Comment (arealclimber) [inline src/components/candlestick_chart/candlestick_chart.tsx:1168]: - return 移除邏輯

File: src/components/candlestick_chart/candlestick_chart.tsx [hunk 3]
```diff
@@ -433,59 +1105,481 @@ export default function CandlestickChart({
...
+                      fontFamily: 'barlow',
+                      fontSize: 12,
+                      // fontWeight: 500,
+                      // stroke: 'black',
+                      // backgroundColor: EXAMPLE_BLUE_COLOR,
+                      // borderBlockColor: EXAMPLE_BLUE_COLOR,
+                      // borderColor: EXAMPLE_BLUE_COLOR,
 
-  // const displayedChart = showPositionOnChart ? () : ()
+                      fill: (d: any) =>
```
Comment (arealclimber) [inline src/components/candlestick_chart/candlestick_chart.tsx:1266]: - hover 標籤要半透明，或者OHLC直接顯示在左上角

**Review by Luphia — COMMENTED** [review]
- fix these issues

**Review by Luphia — APPROVED** [review]
- no bug found

**Pull request merged**